Feature/login thunk

jaceksupernak opened the pull request:
### Feature login thunk
- create login thunk
- refactor signup page to support login through login thunk
- refactor chats page to support logged in person

jaceksupernak committed:
create loginUser thunk

jaceksupernak committed:
refactor chats: turn chatsData to null

jaceksupernak committed:
fix comment

jaceksupernak committed:
change the initial received value of friends to empty array instead of null

jaceksupernak committed:
add comment

KhadidjaArezki reviewed (changes requested):
Since the login and signup routers are done, I think it's time to test integration with the backend.
Can you please uncomment the request login and remove the mock data?
Also, please add a refresh token to the user state and login aaction.

jaceksupernak committed:
remove mock data from the userSlice

jaceksupernak commented:
> Since the login and signup routers are done, I think it's time to test integration with the backend. Can you please uncomment the request login and remove the mock data? Also, please add a refresh token to the user state and login aaction.

I have uncommented the request login, added a refresh token to the user state and login action, removed mock data.

KhadidjaArezki reviewed (approved):
Thanks for the hard work. I hope this was not a rough introduction to redux. Merging.
